### DISCUSSION TITLE
Is it possible to attach multiple images to a Site at once?

### DISCUSSION THREAD

**Discussion by dkischkower-ww-zz** [netbox-community/netbox#6539] [Q&A]
Hey!

Does anyone know if it's currently possible to attach multiple images to a Netbox Site at a time, either via GUI or API?  It seems that this can only be done one image at a time but I wanted to confirm.  We have hundreds of images per Site that we are looking to attach, so to do this one at a time would be incredibly time consuming.

**Comment by candlerb** [comment]
If you click the "API" link at the bottom of the home page, it renders the API documentation (slowly).

From there you'll see it's possible to POST to /api/extras/image-attachments.  By supplying content_type and object_id you should be able to link to a specific site.

However I haven't tested this, and you may have to copy the image itself by hand to the /opt/netbox/netbox/media/image-attachments/ directory.

The API does allow bulk creation: you POST a list of objects, instead of a single object. See https://netbox.readthedocs.io/en/stable/rest-api/overview/#creating-multiple-objects
